transmart-core-db-tests/src/integration-test/groovy/org/transmartproject/db/multidimquery/QueryServiceSpec.groovy

```diff
@@ -374,4 +380,56 @@ class QueryServiceSpec extends TransmartSpecification {
         result*.value.sort() == expectedObservations*.value.sort()
     }
 
+    void "test_study_selection_constraint"() {
+        setupHypercubeData()
+
+        def testObservation = hypercubeTestData.clinicalData.longitudinalClinicalFacts[-1]
+        Constraint constraint = new SubSelectionConstraint(
+                dimension: DimensionImpl.STUDY,
+                constraint: new ValueConstraint(
+                        valueType: "STRING",
+                        operator: Operator.EQUALS,
+                        value: testObservation.textValue
+                )
+        )
+
+        when:
+        def result = multiDimService.retrieveClinicalData(constraint, accessLevelTestData.users[0]).asList()
+        def expectedObservations = hypercubeTestData.clinicalData.longitudinalClinicalFacts
+
+        then:
+        result.size() == expectedObservations.size()
+        result.every { it[DimensionImpl.STUDY] == testObservation.trialVisit.study }
+        result*.value.sort() == expectedObservations*.value.sort()
+    }
+
+    @Ignore("H2 does not support tuple comparisons with IN, which is used for subselections on visits, so this functionality will only work with Postgres or Oracle (Error: Subquery is not a single column query)")
```

Comment:
So maybe it is better to add test like this to transmart-core-db tests, which are running against postgres